src/modes/install.ts

```diff
@@ -65,7 +95,7 @@ export default async function(pkgs: PackageRequirement[]) {
             if (done) {
               throw new PkgxError(`${f} already exists and is not a pkgx installation`)
             }
-            const found = value.match(/^\s*exec pkgx \+([^ ]+)/)?.[1]
+            const found = value.match(/^\s*pkgx \+([^ ]+)/)?.[1]
```

Comment:
this might be dangerous, since it would catch a script with `#!/usr/bin/env pkgx +python ...`

Reply:
Is there some other way of detecting both normal and unsafe packages? We can use special regex to match maybe both `exec pkgx` OR some other unsafe-specific (non-`pkgx`) match?

Reply:
in this instance. at least, you could search for `${pkgstr}.env`. that should be unique to your shim.

Reply:
What if instead, unsafe installations contain a specific line (like `PKGX_UNSAFE=1`) and check that? Unlike safe (normal) installations, we have full control over how the scripts are gonna be, and there's no backwards compatibility requirement.

Reply:
i meant that the script should contain the string `${pkgstr}.env`, since they source it.

Reply:
Yeah, that could be done. Probably with regex as well.